docs/source/design/huggingface_integration.rst

```diff
@@ -0,0 +1,39 @@
+Integration with HuggingFace
+===================================
+
+This document describes how vLLM integrates with HuggingFace libraries. We will explain step by step what happens under the hood when we run ``vllm serve``.
+
+Let's say we want to serve the popular llama model by running ``vllm serve meta-llama/Llama-3.1-8B``.
+
+- The ``model`` argument is ``meta-llama/Llama-3.1-8B``. vLLM will first try to locate the config file ``config.json`` using this argument. See the `code snippet <https://github.com/vllm-project/vllm/blob/10b67d865d92e376956345becafc249d4c3c0ab7/vllm/transformers_utils/config.py#L75>`__ for the implementation.
+
+   - If the ``model`` argument is a local path, vLLM will directly read the config file from the path.
+
+   - Otherwise, vLLM will try to read the config from the HuggingFace cache. See `their website <https://huggingface.co/docs/huggingface_hub/en/package_reference/environment_variables#hfhome>`__ for more information on how the HuggingFace cache works. Here, we can also use the argument ``--revision`` to specify the revision of the model in the cache.
+
+   - If neither of the above works, vLLM will download the config file from the HuggingFace model hub, using the ``model`` argument as the model name, the ``--revision`` argument as the revision, and the environment variable ``HF_TOKEN`` as the token to access the model hub. In our case, vLLM will download the ``config.json`` file from ``https://huggingface.co/meta-llama/Llama-3.1-8B/blob/main/config.json``.
```

Comment:
Perhaps let's not use a gated model for this example? This link will give 404 if the user is not logged in with permission to this model.

Reply:
good point, changed to [05f3374](https://github.com/vllm-project/vllm/pull/10173/commits/05f3374c021a998ace5fd53580f4d4b534a09bf0)